src/components/GitOpsDetailsPage.tsx

```diff
@@ -0,0 +1,61 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+
+import { LoadingBox } from '@patternfly/quickstarts';
+
+import GitOpsDetails from './details/GitOpsDetails';
+import GitOpsEmptyState from './GitOpsEmptyState';
+import { GitOpsEnvironment } from './utils/gitops-types';
+import { getApplicationsBaseURI, getEnvData, getPipelinesBaseURI } from './utils/gitops-utils';
+import useDefaultSecret from './utils/useDefaultSecret';
+import useEnvDetails from './utils/useEnvDetails';
+
+const GitOpsDetailsPage: React.FC<RouteComponentProps<{ appName?: string }>> = ({ match }) => {
+  const { appName } = match.params;
+  const [secretNS, secretName] = useDefaultSecret();
+  const pipelinesBaseURI = getPipelinesBaseURI(secretNS, secretName);
+  const searchParams = new URLSearchParams(location.search);
+  const manifestURL = searchParams.get('url');
+  const applicationBaseURI = getApplicationsBaseURI(appName, secretNS, secretName, manifestURL);
+  const [envs, emptyStateMsg] = useEnvDetails(appName, manifestURL, pipelinesBaseURI);
```

Comment:
Were you able to pass these down to from the Tabs page?  We will be calling the API twice, once here and once in the Tabs page.

Reply:
No, I couldn't pass it for same reason that `HorizontalNavProps` doesn't have customData property. How do you think about we store/cache data and use it from the store?